sympy/physics/units/dimensions.py

```diff
@@ -189,7 +189,7 @@ def __mul__(self, other):
                 other = Dimension(Quantity.get_dimensional_expr(other))
             if isinstance(other, Dimension):
                 return Dimension(self.name*other.name)
-            if not other.free_symbols:
+            if not other.free_symbols: # other.is_number cannot be used
```

Comment:
Why not?

Reply:
No idea. Some test fails.

Reply:
I think knowing which test fails so I can check it would be helpful to me in reviewing this.

Reply:
It was in the first commit:

```
_______ sympy/physics/units/tests/test_util.py:test_dim_simplify_dimless _______
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/sympy-1.4.dev0-py3.6.egg/sympy/utilities/runtests.py", line 1306, in _timeout
    function()
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/sympy-1.4.dev0-py3.6.egg/sympy/physics/units/tests/test_util.py", line 58, in test_dim_simplify_dimless
    assert dim_simplify(sin(L*L**-1)**2*L).get_dimensional_dependencies()\
AttributeError: 'Mul' object has no attribute 'get_dimensional_dependencies'
```
Since it related to units and dimensions, I guessed it was that instance and it looks like I was correct.